### PR TITLE
Cc 682 add healthcheck endpoint

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/CC-682

Add new controller, router and middleware check to skip auth checks if this endpoint is called.

### CODE REVIEW

File: src/controllers/healthcheck.controller.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Request, Response } from "express";
+import config from "../config";
+import { createLogger } from "@companieshouse/structured-logging-node";
+
+const logger = createLogger(config.applicationNamespace);
+
+export const get = (req: Request, res: Response) => {
+    logger.debug(`GET healthcheck`);
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/controllers/healthcheck.controller.ts:8]: or create a variable for the 302, that would be better if the linter accepts it

**Comment by mwestacott** [comment]
Please could you add a link to the JIRA sub-task in the main commit message above.

File: src/controllers/healthcheck.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattch1) [reply src/controllers/healthcheck.controller.ts]: Updated

File: src/controllers/healthcheck.controller.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Request, Response } from "express";
```
[the remark this answers is not in the document]
Comment (mattch1) [reply src/controllers/healthcheck.controller.ts]: Name update

**Review by markpit — APPROVED** [review]

**Pull request merged**